contracts/Lotto.sol

```diff
@@ -0,0 +1,287 @@
+            winningNumbers[i] = i;
+        }
+        allLotteries_[_lottoID].winningNumbers = winningNumbers;
+    }
+
+    /**
+     * @param   _prizeDistribution An array defining the distribution of the 
+     *          prize pool. I.e if a lotto has 5 numbers, the distribution could
+     *          be [5, 10, 15, 20, 30] = 100%. This means if you get one number
+     *          right you get 5% of the pool, 2 matching would be 10% and so on.
+     * @param   _prizePoolInCake The amount of Cake available to win in this 
+     *          lottery.
+     * @param   _startingBlock The block timestamp for the beginning of the 
+     *          lottery. 
+     * @param   _closingBlock The block timestamp after which no more tickets
+     *          will be sold for the lottery. Note that this timestamp MUST
+     *          be after the starting block timestamp. 
+     * @param   _endBlock The block timestamp for the end of the lottery. After
+     *          this time users can withdraw any winnings. Note that between
+     *          the closing block and this end block the winning numbers must
+     *          be added by the admin. If they are not this end block timestamp
+     *          will be pushed back until winning numbers are added. 
+     */
+    function createNewLotto(
+        uint8[] calldata _prizeDistribution,
+        uint256 _prizePoolInCake,
+        uint256 _costPerTicket,
+        uint256 _startingBlock,
+        uint256 _closingBlock,
+        uint256 _endBlock
+    )
+        public
+        onlyOwner()
+        returns(uint256 lottoID)
+    {
+        lotteryIDCounter_.increment();
+
+        allLotteries_[lotteryIDCounter_.current()].lotteryID = lotteryIDCounter_.current();
+        allLotteries_[lotteryIDCounter_.current()].lotteryStatus = Status.NotStarted;
+        allLotteries_[lotteryIDCounter_.current()].prizePoolInCake = _prizePoolInCake;
+        allLotteries_[lotteryIDCounter_.current()].costPerTicket = _costPerTicket;
+        allLotteries_[lotteryIDCounter_.current()].prizeDistribution = _prizeDistribution;
+        allLotteries_[lotteryIDCounter_.current()].startingBlock = _startingBlock;
+        allLotteries_[lotteryIDCounter_.current()].closingBlock = _closingBlock;
+        allLotteries_[lotteryIDCounter_.current()].endBlock = _endBlock;
+        // Emitting important information around new lottery.
+        emit newLotteryCreated(
+            lotteryIDCounter_.current(),
+            Status.NotStarted,
+            _prizeDistribution,
+            _prizePoolInCake,
+            _costPerTicket,
+            _startingBlock,
+            _closingBlock,
+            _endBlock,
+            msg.sender
+        );
+    }
+
+    function check(uint256 _numberOfTickets) public view returns(uint256) {
+        uint256 numberCheck = _numberOfTickets*sizeOfLottery_;
+        return numberCheck;
+    }
+
+    function batchBuyLottoTicket(
+        uint256 _lotteryID,
```

Comment:
you can buy for old lotteries (winning by knowing the tickest, then)

Reply:
Check added